crates/cairo-profiler/src/trace_reader.rs

```diff
@@ -27,14 +23,21 @@ impl From<String> for MeasurementUnit {
 #[derive(Debug, Clone)]
 pub struct MeasurementValue(pub i64);
 
-pub struct ContractCallSample {
-    pub call_stack: Vec<FunctionName>,
+pub struct Sample {
+    pub call_stack: Vec<Function>,
     pub measurements: HashMap<MeasurementUnit, MeasurementValue>,
 }
 
-impl ContractCallSample {
+#[derive(PartialEq, Eq, Hash, Clone)]
+pub enum Function {
+    Entrypoint(FunctionName),
+    _Inlined(FunctionName),
+    NonInlined(FunctionName),
+}
```

Comment:
Not a good naming scheme. 
```
    _Inlined(FunctionName),
    NonInlined(FunctionName),
```
those form a complete set, should cover all elements of the type.

I would suggest instead
```
pub enum Function {
    Entrypoint(FunctionName),
    InternalFunction(InternalFunction),
}

pub enum InternalFunction {
    _Inlined(FunctionName),
    NonInlined(FunctionName),
}
```

Reply:
Good idea, `Entrypoint` is used the same way as `NonInlined` func after that, but I wanted to distinct it since it is more readable